primary/src/block_synchronizer/mod.rs

```diff
@@ -429,7 +432,7 @@ impl<PublicKey: VerifyingKey> BlockSynchronizer<PublicKey> {
         let key = RequestID::from_iter(to_sync.iter());
 
         let message = PrimaryMessage::<PublicKey>::CertificatesBatchRequest {
-            certificate_ids: block_ids,
+            certificate_ids: to_sync.clone(),
```

Comment:
A nasty bug that discovered during end to end testing in `get_collections` in gRPC.